### PR TITLE
feat: :sparkles: specify `output_dir` in reference scripts

### PR DESCRIPTION
**Pull request by odulcy-mindee**

### CODE REVIEW

File: references/detection/train_tensorflow.py [hunk 1]
```diff
@@ -357,7 +358,7 @@ def main(args):
             min_loss = val_loss
         if args.save_interval_epoch:
             print(f"Saving state at epoch: {epoch + 1}")
-            model.save_weights(f"./{exp_name}_{epoch + 1}.weights.h5")
+            model.save_weights(Path(args.output_dir) / f"{exp_name}_{epoch + 1}.weights.h5")
```
Comment (felixT2K) [inline references/detection/train_tensorflow.py:361]: @odulcy-mindee missing above line 357:

model.save_weights(f"./{exp_name}.weights.h5")

File: references/detection/train_pytorch_ddp.py [hunk 2]
```diff
@@ -410,11 +411,11 @@ def main(rank: int, world_size: int, args):
             )
             if val_loss < min_loss:
                 print(f"Validation loss decreased {min_loss:.6} --> {val_loss:.6}: saving state...")
-                torch.save(model.module.state_dict(), f"./{exp_name}.pt")
+                torch.save(model.module.state_dict(), Path(args.output_dir) / f"{exp_name}.pt")
                 min_loss = val_loss
             if args.save_interval_epoch:
                 print(f"Saving state at epoch: {epoch + 1}")
-                torch.save(model.state_dict(), f"./{exp_name}_epoch{epoch + 1}.pt")
+                torch.save(model.state_dict(), Path(args.output_dir) / f"{exp_name}_epoch{epoch + 1}.pt")
```
Comment (felixT2K) [inline references/detection/train_pytorch_ddp.py:418]: model.module.state_dict for ddp right ? ^^

File: references/classification/train_pytorch_character.py
Comment (felixT2K) [inline references/classification/train_pytorch_character.py:1]: missing

parser.add_argument("--output_dir", type=str, default=".", help="path to save checkpoints and final model")

here :)

**Review by felixT2K — CHANGES_REQUESTED** [review]
Small things :)

**Comment by odulcy-mindee** [comment]
@felixT2K Thanks for the catch

**Review by felixdittrich92 — APPROVED** [review]
Thx ☺️ lgtm

**Pull request merged**